Sources/MapboxSearch/PublicAPI/Engine/SearchEngine.swift

```diff
@@ -273,6 +273,20 @@ public class SearchEngine: AbstractSearchEngine {
         return SearchResponse(coreResponse: coreResponse)
     }
     
+    private func preProcessRetrieveResponse(_ coreResponse: CoreSearchResponseProtocol?) -> SearchResponse? {
+        assert(Thread.isMainThread)
+        
+        guard let coreResponse = coreResponse else {
+            assertionFailure("Response should never be nil")
+            eventsManager.reportError(.responseProcessingFailed)
+            delegate?.searchErrorHappened(searchError: .responseProcessingFailed, searchEngine: self)
+
+            return nil
+        }
+        
```

Comment:
So you remove both `queryValue` cases for retrieval:
1. `coreResponse.request.query != query` - which is the root cause of the bug (okay here) 
2. but for category is it okay to remove this too `!coreResponse.request.query.isEmpty`?

Reply:
Got it, with `suggestion` already available we do not need both (and any) check on it.